test/e2e/setup.go

```diff
@@ -73,6 +83,151 @@ func skipIfNotInDevelopmentEnv() {
 	}
 }
 
+func TakeScreenshot(wd WebDriver, e error) {
+	log.Infof("Error : %s", e.Error())
+	log.Info("Taking Screenshot and saving page source")
+	imageBytes, err := wd.Screenshot()
+	if err != nil {
+		panic(err)
+	}
+
+	imageData, err := png.Decode(bytes.NewReader(imageBytes))
+	if err != nil {
+		panic(err)
+	}
+
+	sourceString, err := wd.PageSource()
+	if err != nil {
+		panic(err)
+	}
+
+	errorString := strings.ReplaceAll(e.Error(), " ", "_")
+
+	imagePath := "./" + errorString + ".png"
+	sourcePath := "./" + errorString + ".html"
+
+	imageAbsPath, err := filepath.Abs(imagePath)
+	if err != nil {
+		panic(err)
+	}
+	sourceAbsPath, err := filepath.Abs(sourcePath)
+	if err != nil {
+		panic(err)
+	}
+
+	image, err := os.Create(imageAbsPath)
+	if err != nil {
+		panic(err)
+	}
+
+	source, err := os.Create(sourceAbsPath)
+	if err != nil {
+		panic(err)
+	}
+
+	err = png.Encode(image, imageData)
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = source.WriteString(sourceString)
+	if err != nil {
+		panic(err)
+	}
+
+	err = image.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	err = source.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	log.Infof("Screenshot saved to %s", imageAbsPath)
+	log.Infof("Page Source saved to %s", sourceAbsPath)
+
+	panic(e)
+}
+
+func adminPortalSessionSetup() (string, *WebDriver) {
+	const (
+		port = 4444
+	)
+
+	os.Setenv("SE_SESSION_REQUEST_TIMEOUT", "9000")
+
+	caps := Capabilities{
+		"browserName":         "MicrosoftEdge",
+		"acceptInsecureCerts": true,
+	}
+	wd := WebDriver(nil)
+
+	var err error
+
+	_, err = url.ParseRequestURI("https://localhost:4444")
```

Comment:
Nitpicking: we could just initialize it in one go. If you have some more commits to push , it would be great to include it, otherwise I think we can let it pass

Reply:
Fixed!